### ISSUE TITLE
PaymentModule - Multiple Orders for a Cart

### ISSUE THREAD

**Issue by Matt75** [PrestaShop/docs#508]
Sometime PrestaShop will split a Cart in multiple Order with same Reference after a Payment.

This is made in [PaymentModule::validateOrder()](https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/PaymentModule.php#L257)

PrestaShop will send to PaymentModule only [Order id](https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/PaymentModule.php#L31) and [Order reference](https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/PaymentModule.php#L32) of the last [created Order](https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/PaymentModule.php#L712).

Every PaymentModule will only update OrderState of the last created Order, every child Order should be manually update by merchant because he needs to choose if additional shipping cost should be paid by Customer or if he take it in charge.

Need more informations to find every possible cases will cause Order splitting and how to handle that for Module developer.

**Comment by jf-viguier** [comment]
For me, the best and easy way should be to remove order splitting. It's not needed anymore and upset ERP and accountants

**Comment by msaustral** [comment]
We have one customer with same issue PS 1.7.6.9

There is something strange, on product -> transportation -> weight, the decimals separator is a coma (,) but in the total wight of the order it is shown in gram but it say Kg in text.

Example:

On product:
![image](https://user-images.githubusercontent.com/46203943/120585704-3f8a3b80-c400-11eb-929e-2a65d6fe4b97.png)


Total weight on order:

![image](https://user-images.githubusercontent.com/46203943/120585427-ba068b80-c3ff-11eb-8e98-aa4cd073e6b6.png)

On carrier
![image](https://user-images.githubusercontent.com/46203943/120585773-5fb9fa80-c400-11eb-92d4-4401c2a67abd.png)


it should show 12,430 kg but it say 12.430 Kg, if the PS taking the weight in grams to calculate shipping cost, it will go out of carrier rage and split the order.

in this case we have a rage of 60 Kg and the order will send twelve thousand four hundred thirty (12.430) Kg

**Comment by loreactm** [comment]
Hello, any updates on this issue? Randomly we get orders with one product split into a linked order. None of these products have carrier exceptions and they fit within the courier weight setting.. so I'm not sure what's causing it.

Is there a way to disable the linked orders feature completely?

**Comment by msaustral** [comment]
Hi any news on this case?, we do have multiples customers affected with this.

**Comment by Lionel-dev** [comment]
Hi, we have news on this, I have some customers being impacted and I do not know how to justify this bug to them...

**Comment by wassafr** [comment]
Also happens to us with no case listed above.

**Comment by Matt75** [comment]
> Also happens to us with no case listed above.

Yes, but as I said it should be completed, please share we us your use cases ;)

**Comment by sloofa** [comment]
This issue still remains also in PS8?
I have bought Webkul module for split order and only first order is sent to PaymentModule

**Comment by jouino16** [comment]
I have just shared my humble opinion on the split order case here https://github.com/PrestaShop/prestashop-specs/issues/111#issuecomment-2802706738.

Some of what I shared in my comment is also regarding the issue mentioned in this thread, which is the fact that payment modules are not working correctly out of the box in the context of split order.

In my humble opinion, this is normal, as all the payment and shipping modules are implemented for the classic context, not the split order context. So they do not handle this out of the box. So when setting up a store with split order context, the developer has to adapt each involved payment or shipping module (especially those with pickup locations) to the split order context (please check section "6. Recommendations" in the documentation mentioned below). For example, I tested the module I developed (more on this below) with Stripe (v3.1.4) and PayPal (v6.0.2) which updated all sub-orders with the status "Payment" accepted, but it was not the case with Revolut.

Regarding the module I developed (mentioned in the previous paragraph). It is [Multi Shipping / Multiple Carriers Order](https://addons.prestashop.com/en/shipping-carriers/93768-multi-shipping-multiple-carriers-order.html) module (also available [here](https://smartunicorn.fr/produit/module-prestashop-commande-avec-transporteurs-livraisons-multiples/) - **check for better FAQ section readability**). In a nutshell, it enables multi shipping / multi carrier orders in PrestaShop. The order will be split based on the number of the carriers. Here is [the demo video](https://www.youtube.com/watch?v=5QQMzO3-EdI&ab_channel=SmartUnicorn).

Here is the module documentation [Multi Shipping - Multiple Carriers Order_documentation.pdf](https://github.com/user-attachments/files/19742155/Multi.Shipping.-.Multiple.Carriers.Order_documentation.pdf) where you can find examples and the limitations to be aware of.

**Regarding the wrong weight calculation**: this is also an expected bug. It is happening because the calculations are based on the products listed in the cart (meaning the total ordered products). The solution is to make the shipping calculation based only on the products to be shipped by the current carrier used at the sub-order level.
Example, if we have a split order with 2 sub-orders as follows:
- sub-order 1 with product A shipped by carrier C1: only product A should be involved in carrier C1 shipping cost.
- sub-order 2 with product B shipped by carrier C2: only product B should be involved in carrier C2 shipping cost.

Since there are 2 distinct shipments, there will be 2 distinct shipping costs (one per sub-order).

This has been handled by the above-mentioned module I developed.

Hope this helps!
